tests/test_builder.py

```diff
@@ -579,6 +582,246 @@ def test_try():
     )
 
 
+def test_while_true():
+    with DebugConnection("Alice") as conn:
+
+        with conn.while_true(max_iterations=10) as loop:
+            q = Qubit(conn)
+            m = q.measure()
+            constraint = ValueAtMostConstraint(m, 42)
+            loop.set_exit_condition(constraint)
+
+        subroutine = conn.builder.subrt_pop_pending_subroutine()
+        print(subroutine)
```

Comment:
I see a nice program with the while construction, but the test doesn't validate to true or false. When this test is run as a single test I get a 'TypeError: calling <function test_while_true at 0x0000021F556D7708> returned None, not a test'. Maybe check if the subroutine is defined correctly by checking all or parts of the commands list. Edit: I see a nice PreSubroutineInspector used in other tests.

Reply:
How are you running the test? When running `pytest tests/test_builder.py -k while_true` the test passes for me. In any case I also added a pattern match test.

Reply:
I run it as a single test.. the pattern match makes it a test with result